api/resources_portal/views/attachment.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import serializers, viewsets
+from rest_framework.permissions import BasePermission, IsAdminUser, IsAuthenticated
+from rest_framework.response import Response
+
+from guardian.shortcuts import assign_perm
+
+from resources_portal.models import Attachment, MaterialRequest
+from resources_portal.views.relation_serializers import MaterialRelationSerializer
+
+
+class AttachmentSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Attachment
+        fields = (
+            "id",
+            "filename",
+            "description",
+            "s3_bucket",
+            "s3_key",
+            "deleted",
+            "created_at",
+            "updated_at",
+            "sequence_map_for",
+        )
+        read_only_fields = ("id", "created_at", "updated_at")
+
+
+class AttachmentDetailSerializer(AttachmentSerializer):
+    sequence_map_for = MaterialRelationSerializer(many=False, read_only=True)
+
+
+class HasViewAttachmentOrIsAdminUser(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return request.user.has_perm("view_attachment", obj) or request.user.is_staff
+
+
+class HasModifyAttachmentOrIsAdminUser(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return request.user.has_perm("modify_attachment", obj) or request.user.is_staff
```

Comment:
Yeah when I see this here, I think we want these to be organization level, not request level, but I await @dvenprasad's confirmation.

Reply:
Yes, we want the permissions on an organization level